paddle/fluid/operators/center_loss_op.cu

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <iostream>
+#include "paddle/fluid/operators/center_loss_op.h"
+#include "paddle/fluid/platform/cuda_primitives.h"
+#include "paddle/fluid/platform/gpu_info.h"
+namespace paddle {
+namespace operators {
+
+using platform::PADDLE_CUDA_NUM_THREADS;
+
+template <typename T>
+__global__ void ComputeDistance(int64_t elements_num, const T *X,
+                                const int x_width, const int64_t *label,
+                                const T *centers_data, T *diff_xc) {
+  for (int index = blockIdx.x * blockDim.x + threadIdx.x; index < elements_num;
+       index += blockDim.x * gridDim.x) {
+    int row = index / x_width;
+    int col = index % x_width;
+    int64_t center_idx = label[row];
+    diff_xc[index] = X[index] - centers_data[center_idx * x_width + col];
+  }
+}
+
+template <typename T>
+__global__ void UpdateCenters(int cluster_num, const int samples_num,
+                              const int x_width, const int64_t *label,
+                              const T *diff_xc, T *acc_sum, const T *alpha,
+                              T *centers_out) {
+  for (int64_t cidx = blockIdx.x * blockDim.x + threadIdx.x; cidx < cluster_num;
```

Comment:
low efficiency. A solution is given in the next comment

Reply:
done